src/core/net/ip6_mpl.cpp

```diff
@@ -406,6 +406,8 @@ void Mpl::HandleRetransmissionTimer(void)
                         messageCopy->SetSubType(Message::kSubTypeMplRetransmission);
                     }
 
+                    messageCopy->SetAllowLoopbackToHost(true);
+                    messageCopy->SetOrigin(Message::kOriginHostTrusted);
```

Comment:
Can we use the values from` Clone()` from original message?

Reply:
I think changing the origin here in `ip6_mpl` module and  `HandleRetransmissionTimer()` is ok since we are sending a multicast msg that we have received with its origin being `kOriginThread`.

But I think we should not need to update the origin in `Ip6::SendDatagram()` since from what I see `SendDatagram()` is used with a new message we constructed?

Reply:
You're right. Seems we don't need to re-set the Origin anywhere in `ip6.cpp`.